utils/merkletrie/fsnoder/file.go

```diff
@@ -0,0 +1,73 @@
+package fsnoder
+
+import (
+	"bytes"
+	"fmt"
+	"hash/fnv"
+
+	"gopkg.in/src-d/go-git.v4/utils/merkletrie/noder"
+)
+
+// file values represent file-like noders in a merkle trie.
+type file struct {
+	name     string // relative
+	contents string
+	hash     []byte // memoized
+}
+
+// newFile returns a noder representing a file with the given contents.
+func newFile(name, contents string) (*file, error) {
+	if name == "" {
+		return nil, fmt.Errorf("files cannot have empty names")
+	}
+
+	return &file{
+		name:     name,
+		contents: contents,
+	}, nil
+}
+
+// The hash of a file is just its contents.
+// Empty files will have the fnv64 basis offset as its hash.
+func (f *file) Hash() []byte {
+	if f.hash == nil {
+		h := fnv.New64a()
+		h.Write([]byte(f.contents))
+		f.hash = []byte{}
+		f.hash = h.Sum(f.hash)
+	}
+
+	return f.hash
+}
+
+func (f *file) Name() string {
+	return f.name
+}
+
+func (f *file) IsDir() bool {
+	return false
+}
+
+func (f *file) Children() ([]noder.Noder, error) {
+	return noder.NoChildren, nil
```

Comment:
`noder.NoChildren` seems to suggest that the result is a noder with no children, while it is actually an empty slice of nodes (i.e. `nil`). Shouldn't this be just `return nil, nil`?